addons/website/models/website.py

```diff
@@ -872,40 +872,16 @@ def _handle_views_and_pages(views):
 
     def _get_alternate_languages(self, canonical_params):
         self.ensure_one()
-
-        if not self._is_canonical_url(canonical_params=canonical_params):
-            # no hreflang on non-canonical pages
-            return []
```

Comment:
Why removing this?
I tested on your branch and it seems to be breaking that:

I am on an alias domain `https://24739484-master.runbot173.odoo.com`
The canonical url is `https://24739484-master-all.runbot173.odoo.com` (see the `-all` part)
![image](https://user-images.githubusercontent.com/30048408/223191859-3e48220a-7f49-433d-b981-1eb3034c6fb3.png)

But I still see the alternate tags

Reply:
I removed it for two reasons:
- I need to return the data for the language selector --> no speed gain
- I don't see any reason to remove alternates on non-canonical URLs; so it's easier to always do it.

Reply:
According to the introducing commit https://github.com/odoo/odoo/commit/1781041f13b66dfe91b3230794ffb6b2001d86c1
```
Adapt hreflang, because it:
- must only be present on canonical pages
- must always lead to canonical pages
- should not be set if there is no alternate language
```
While you seem to still respect second point, you don't respect first point anymore.
I did not dig into SEO articles to confirm it was the case but I am pretty sure SEB did this for a good reason.
I'd keep it.